File: WorkIt/app/src/main/java/prot3ct/workit/data/remote/base/AuthDataContract.java
package prot3ct.workit.data.remote.base;

import io.reactivex.Observable;

public interface AuthDataContract {
    Observable<Boolean> login(String username, String password);

    Observable<Boolean> register(String email, String fullName, String password);

    Observable<Boolean> autoLogin();

    int getLoggedInUserId();

    String getLoggedInUserName();

    void logoutUser();

    boolean isLoggedIn();
}

File: WorkIt/app/src/main/java/prot3ct/workit/data/remote/base/AuthDataContract.kt
package prot3ct.workit.data.remote.base

import io.reactivex.Observable

interface AuthDataContract {
    fun login(username: String, password: String): Observable<Boolean>

    fun register(email: String, fullName: String, password: String): Observable<Boolean>

    fun autoLogin(): Observable<Boolean>

    val loggedInUserId: Int

    val loggedInUserName: String

    fun logoutUser()

    val isLoggedIn: Boolean
}